sklearn_numba_dpex/kmeans/kernels.py

```diff
@@ -1,942 +0,0 @@
-import math
-from functools import lru_cache
-
-import dpctl
-import numba_dpex as dpex
-
-# General note on kernel implementation
```

Comment:
Is it possible to do another PR which would precedes this PR for the removal of this file and the migration of its content to the files in `sklearn_numba_dpex/kmeans/kernels`? This would reduce the diff of this PR and I think ease reviews (I think having at max ± 500 diffs is a good rule of thumb for faster integration).

Reply:
While I agree with your rule of thumbs for diffs, i don't think it should apply here since the diff was artificially high because of copy/paste, even if they appear in the diff I don't think the internals of the copy/pasted blocks should be reviewed, they have not been edited, only moved left and right.

Even after rebasing on #18 it remains fairly high because of duplicated code that can't be refactored because known limitations with numba_dpex.

Reply:
> While I agree with your rule of thumbs for diffs, i don't think it should apply here since the diff was artificially high because of copy/paste, even if they appear in the diff I don't think the internals of the copy/pasted blocks should be reviewed, they have not been edited, only moved left and right.

I agree, but to me, having blocks of code moved around without changes still hardened reviews. PR like #18 highly help in this regards.